### DISCUSSION TITLE
Investigative analysis using Github CoPilot

### DISCUSSION THREAD

**Discussion by vidyaiyer-link** [community/community#126195] [Programming Help]
### Select Topic Area

Question

### Body

can we have some links, tutorials  to leverage investigative analysis of Java code when our repository is not on Github

*Personal identifying information removed by litebrite82

**Comment by LiteBrite82** [comment]
Hi @vidyaiyer-link ,

I've removed your private information from [this post](https://github.com/orgs/community/discussions/126195#discussion-6741885) as posting identifiable information is not allowed, per our [Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md).

**Comment by LiteBrite82** [comment]
@vidyaiyer-link 

I'm trying to better understand your question. 

I think I understand you to mean: 
You want help finding 3rd party tools or guides to analyze Java code in repos on platforms other than GitHub; ie GitLab? 

If that is correct, can you add some specific parameters you are looking for? What are you trying to accomplish?

**Comment by vidyaiyer-link** [comment]
hi,
 have added some cases and scenarios so that you can understand our queries


Case-1
To reproduce issue which occurred in an output file ( e.g. PDF , CSV etc ), developer need to identify & run batch job or trigger the processes from GUI to identify the flow. In addition to that , correct data set-up is also required in lower envs. Post debugging the flow, developer could pin-point exact piece of code causing issue. Issue could be like few lines are missing in output file or alignment is not proper etc.

Case-2
To identify performance issues for some of the processes/jobs, developer need to brainstorm / collect the stats to find possible bottleneck. This also takes running the processes/batch jobs locally by adding loggers in code flow with timers, analyzing the heap dump. E.g. moving the files from temp folder to destination folder was taking more time so solution identified was to move files from temp folder to destination folder.

**Comment by Sharique55** [comment]
🚀 Just dropped a complete toolkit on Java Heap Dump Analysis using JProfiler, jmap & Eclipse MAT.
Packed with examples, diagrams, and memory leak hunting strategies.
🔍 Check it out, star it, and share: https://github.com/Sharique55/Java-Heap-Dump-Analysis-Toolkit
